### DISCUSSION TITLE
GitHubみたいに文字列選択＋URLペーストでリンクを作れる機能がほしい

### DISCUSSION THREAD

**Discussion by JunichiIto** [increments/qiita-discussions#587] [要望・アイデア・不具合]
GitHubだとテキストエリアの文字列を選択してURLをペーストすると、選択部分をマークダウンのリンクに整形してくれる機能があります。

![chiAEQ3H5s](https://github.com/increments/qiita-discussions/assets/1148320/ef545dec-2dc8-42c7-90e7-5197e8c54bfc)

Qiitaでもあとから「この文字列をリンクにしたい」と思うことはよくあるので、GitHubと同じような機能があるとテキスト編集の手間が減ってありがたいです。

**Comment by tanaka-asami415** [comment]
@JunichiIto 
リクエストありがとうございます🙏

人によって使い心地が変わりそうなため、実現方法は検討したいと思いますが
手間が減らせるよう改善の参考にさせていただきます！

**Reply by P-SiZK** [reply]
@JunichiIto
お待たせしました :bow:
実現方法を検討した結果、リンクを挿入するショートカットを追加することにしました。

https://qiita.com/release-notes/2024-06-20

下記の画像のように `command+K`, `Ctrl+K` でリンクが挿入されます。
- 文字列を選択していない場合 `[title](url)` を挿入
- URL を選択している場合 `url` の部分を選択した URL に置換
- URL 以外を選択している場合 `title` の部分を選択した文字列に置換

![](https://github.com/increments/qiita-discussions/assets/54477535/fbe6fae6-a6b3-4de5-8e2d-a00a3b8c0307)

**Reply by JunichiIto** [reply]
おお、どうもありがとうございます！！
Command-Kでリンクを挿入できることを確認しました。

ただ、ショートカットだとみんな気付きにくいかも？とか思ったりしました。
ショートカットの代わりにメニューバーにリンク用のボタンがあるとさらに良いかもしれません。

![Screenshot 2024-06-21 at 11 46 29](https://github.com/increments/qiita-discussions/assets/1148320/192d2fa4-a830-41d6-a556-8db885bd20ef)

このGitHubのテキストエリアにもリンク用のボタンはありますよね。
![Screenshot 2024-06-21 at 11 47 26](https://github.com/increments/qiita-discussions/assets/1148320/0281b944-104f-45cc-942c-3b768690694d)

ちなみに他にもショートカットってあるんですか？
ショートカット一覧みたいなページがあれば知りたいです。

**Reply by tanaka-asami415** [reply]
@JunichiIto 
ご確認ありがとうございます！

> ただ、ショートカットだとみんな気付きにくいかも？とか思ったりしました。
ショートカットの代わりにメニューバーにリンク用のボタンがあるとさらに良いかもしれません。

はい！メニューバーのボタンについても、新たなショートカットの追加と合わせて将来的に実装予定です。
ボタンの実装と合わせると先になってしまいそうだったため、今回は取り急ぎリンクのショートカットのみ対応いたしました🙇

現時点では、ショートカット一覧はご用意できていないのですが、こちらも合わせて検討いたします。
今ご用意できているショートカットには以下のようなものがございます！

- エディタ検索（Command/Ctrl + F）
- 行の入れ替え（Option/Alt + ↑↓）
- インデント（Tab、Shift + Tab）

**Reply by JunichiIto** [reply]
@tanaka-asami415 返信ありがとうございます。
もろもろ了解です！ご対応どうもありがとうございました。
